visualization/app/codeCharta/ui/maxTreeMapFiles/maxTreeMapFiles.component.ts

```diff
@@ -0,0 +1,34 @@
+import "./maxTreeMapFiles.component.scss"
+import { MaxTreeMapFilesSubscriber, MaxTreeMapFilesService } from "../../state/store/appSettings/maxTreeMapFiles/maxTreeMapFiles.service"
+import { IRootScopeService } from "angular"
+import { StoreService } from "../../state/store.service"
+import { setMaxTreeMapFiles } from "../../state/store/appSettings/maxTreeMapFiles/maxTreeMapFiles.actions"
+
+export class MaxTreeMapFilesController implements MaxTreeMapFilesSubscriber {
+	private _viewModel: {
+		maxTreeMapFiles: number
+	} = {
+		maxTreeMapFiles: null
+	}
+
+	constructor(private $rootScope: IRootScopeService, private storeService: StoreService) {
+		MaxTreeMapFilesService.subscribe(this.$rootScope, this)
+		const maxTreeMapFiles = this.storeService.getState().appSettings.maxTreeMapFiles
+		this.onMaxTreeMapFilesChanged(maxTreeMapFiles)
+		this._viewModel.maxTreeMapFiles = maxTreeMapFiles
```

Comment:
Can you extract that in a function called `initOnOpenDialog()` or similar? Otherwise somebody might remove the lines, as it looks like a duplicate for `onMaxTreeMapFilesChanged()`, but it is needed as a closed (non existing) dialog, does not receive events.

And remove line 18, as this one is a duplicate of the function call in `onMaxTreeMapFilesChanged()` :)